Patch by MSP-Greg
Changes to allow tests to pass when run parallel on Appveyor.
Changes often involve setting timeouts longer to allow disk I/O
diff --git a/test/ruby/test_file_exhaustive.rb b/test/ruby/test_file_exhaustive.rb
index 4c5540eb4c..767a0c16b6 100644
--- a/test/ruby/test_file_exhaustive.rb
+++ b/test/ruby/test_file_exhaustive.rb
@@ -1311,12 +1311,14 @@ def test_truncate
   end
 
   def test_flock_exclusive
+    time_out = !(/mingw|mswin|bccwin/ !~ RUBY_PLATFORM) ? 0.5 : 0.1
     File.open(regular_file, "r+") do |f|
       f.flock(File::LOCK_EX)
       assert_separately(["-rtimeout", "-", regular_file], "#{<<~begin}#{<<~"end;"}")
       begin
         open(ARGV[0], "r") do |f|
-          Timeout.timeout(0.1) do
+          time_out = !(/mingw|mswin|bccwin/ !~ RUBY_PLATFORM) ? 0.5 : 0.1
+          Timeout.timeout(time_out) do
             assert(!f.flock(File::LOCK_SH|File::LOCK_NB))
           end
         end
@@ -1324,8 +1326,9 @@ def test_flock_exclusive
       assert_separately(["-rtimeout", "-", regular_file], "#{<<~begin}#{<<~"end;"}")
       begin
         open(ARGV[0], "r") do |f|
+          time_out = !(/mingw|mswin|bccwin/ !~ RUBY_PLATFORM) ? 0.5 : 0.1
           assert_raise(Timeout::Error) do
-            Timeout.timeout(0.1) do
+            Timeout.timeout(time_out) do
               f.flock(File::LOCK_SH)
             end
           end
@@ -1342,7 +1345,8 @@ def test_flock_shared
       assert_separately(["-rtimeout", "-", regular_file], "#{<<~begin}#{<<~"end;"}")
       begin
         open(ARGV[0], "r") do |f|
-          Timeout.timeout(0.1) do
+          time_out = !(/mingw|mswin|bccwin/ !~ RUBY_PLATFORM) ? 0.5 : 0.1
+          Timeout.timeout(time_out) do
             assert(f.flock(File::LOCK_SH))
           end
         end
@@ -1351,7 +1355,8 @@ def test_flock_shared
       begin
         open(ARGV[0], "r+") do |f|
           assert_raise(Timeout::Error) do
-            Timeout.timeout(0.1) do
+            time_out = !(/mingw|mswin|bccwin/ !~ RUBY_PLATFORM) ? 0.5 : 0.1
+            Timeout.timeout(time_out) do
               f.flock(File::LOCK_EX)
             end
           end
